### PR TITLE
Add Trees to Map

### PR DESCRIPTION
**Pull request by SofieDunt**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/925293934/views/18904377/pulses/926128754)

Allows users to see trees displayed on a map.

## This PR

Visualized `sitesLayer` only shows sites where trees are planted.

Added tree icons to assets
- green for regular trees, blue for young trees
- currently the logic for young trees isn't right, just did this to lay groundwork for when we can get the date the tree was planted and to show that different icons based on feature properties are possible
- used images instead of built-in google maps symbols for icons because they load significantly faster

MapViews enum describes what data layers are visible on the map
- set desired view for the map at the container level (landing = trees, reservations = blocks)

## Screenshots

 
Landing 
(minimum zoom at which trees are visible)
MinZoom
(increased zoom)
Zoom1
Zoom2
(default zoom when searching an address)
ZoomAddress

UPDATE:
Increased minimum zoom breakpoint and using color-blind friendly colors:
image

## Verification Steps

 
Viewed sites on map and checked that it looked reasonable. Randomly selected about 40 of the displayed sites and all of the blue sites and made sure that the site had a tree present and was the appropriate color for the `updated_at` date (did this by adding an event listener that opened an infowindow with content `tree_present` and `updated_at`).

### CODE REVIEW

**Review by hridaysabir — APPROVED** [review]
Looks good to me!

File: src/components/mapPageComponents/mapContent/index.tsx [hunk 1]
```diff
@@ -18,16 +18,21 @@ const EmptyMapContainer = styled.div`
...
 `;
 
-interface MapContentProps {
+interface MapContentStateProps {
   readonly blocks: MapGeoDataReducerState['blockGeoData'];
   readonly neighborhoods: MapGeoDataReducerState['neighborhoodGeoData'];
   readonly sites: MapGeoDataReducerState['siteGeoData'];
 }
 
+interface MapContentProps extends MapContentStateProps {
```
Comment (willmt80) [inline src/components/mapPageComponents/mapContent/index.tsx:27]: Nice!

File: src/components/mapPageComponents/mapView/index.tsx [hunk 2]
```diff
@@ -30,13 +32,27 @@ const MapDiv = styled.div`
   height: 100%;
 `;
 
+// Three years before the current date
+const breakpointDate = new Date().setFullYear(new Date().getFullYear() - 3);
+
+export enum MapViews {
```
Comment (willmt80) [inline src/components/mapPageComponents/mapView/index.tsx:38]: ⛏️   I know it's not explicitly related to redux but we might want to put this in a types file

File: src/components/mapPageComponents/mapView/index.tsx [hunk 3]
```diff
@@ -256,6 +272,38 @@ const MapView: React.FC<MapViewProps> = ({ blocks, neighborhoods, sites }) => {
...
+            }
+
+            return {
+              visible,
+              icon,
+            };
+          });
+        }
+
+        setSitesStyle(false);
```
Comment (willmt80) [inline src/components/mapPageComponents/mapView/index.tsx:305]: Does this help with performance?

File: src/components/mapPageComponents/mapView/index.tsx [hunk 4]
```diff
@@ -256,6 +272,38 @@ const MapView: React.FC<MapViewProps> = ({ blocks, neighborhoods, sites }) => {
...
+            let visible = false;
+            let icon = treeIcon;
+
+            // Only shows sites if there is a tree there
+            if (feature.getProperty('tree_present')) {
+              visible = v && true;
+            }
+
+            // TODO: update this to if the tree was planted within the past three years
+            // If the tree has not been updated within the past three years, use youngTreeIcon
```
Comment (willmt80) [inline src/components/mapPageComponents/mapView/index.tsx:292]: Right now this is a png. Are we going to have different images for the different adoption statuses as well?

**Review by willmt80 — APPROVED** [review]
💯 Good work on this! Left a couple small comments and questions

File: src/components/mapPageComponents/mapView/index.tsx (hunk 3 above)
Comment (SofieDunt) [reply src/components/mapPageComponents/mapView/index.tsx:305]: This sets the default visibility as false, so when you first load the map you can't see the sites (visibility becomes true after zooming in).

File: src/components/mapPageComponents/mapView/index.tsx (hunk 4 above)
Comment (SofieDunt) [reply src/components/mapPageComponents/mapView/index.tsx:292]: Yeah, I'll add that now

**Pull request merged**